pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1553,3 +1557,21 @@ func protoEncodeFlowlogCounters(counters types.FlowlogCounters) *metrics.Flowlog
 		FailedAttempts: counters.FailedAttempts,
 	}
 }
+
+// getDormantTime returns scaled dormant time
+func getDormantTime(ctx *zedagentContext) uint64 {
+	return uint64(ctx.globalConfig.GlobalValueInt(types.MetricInterval) * ctx.dormantTimeScaleFactor)
+}
+
+// scaleDormantTimeScaleFactor scale's dormantTimeScaleFactor up or down (depending upon scaleUp)
+// by metricsFailureScaleFactor. Updated dormantTimeScaleFactor should  always be
+// >= minDormantTimeScaleFactor && <= maxDormantTimeScaleFactor
+func scaleDormantTimeScaleFactor(ctx *zedagentContext, scaleUp bool) {
+	if scaleUp {
+		ctx.dormantTimeScaleFactor = uint32(math.Ceil(
+			math.Min(float64(ctx.dormantTimeScaleFactor*metricsFailureScaleFactor), maxDormantTimeScaleFactor)))
+	} else {
+		ctx.dormantTimeScaleFactor = uint32(math.Ceil(
+			math.Max(float64(ctx.dormantTimeScaleFactor/metricsFailureScaleFactor), minDormantTimeScaleFactor)))
```

Comment:
Seems very clunky to convert to float and back given that this is all integers.

Reply:
I removed this from this PR, but your comment is noted and will be addressed in the PR for dynamic adjustment.